app.py

```diff
@@ -320,6 +320,8 @@ def parse_url_params():
         "Select a channel:",
         CHANNELS,
         key="channel",
+        # Use the user provided channel (via query params) if possible.
+        index=url_params["channel"] in CHANNELS and CHANNELS.index(url_params["channel"]) or 0,
```

Comment:
I prefer this ternary form:

```suggestion
        index=CHANNELS.index(url_params["channel"]) if url_params["channel"] in CHANNELS else 0,
```